perceval/converters/myqlm_converter.py

```diff
@@ -0,0 +1,151 @@
+        # this nested import fixes automatic class reference generation
+
+        # count the number of CNOT gates to use during the conversion, will give us the number of herald to handle
+        n_cnot = qlmc.count("CNOT")
+        cnot_idx = 0
+
+        n_moi = qlmc.nbqbits * 2  # number of modes of interest = 2 * number of qbits
+        input_list = [0] * n_moi
+        p = Processor(self._backend_name, n_moi, self._source)
+
+        for i in range(qlmc.nbqbits):
+            p.add_port(i * 2, Port(Encoding.DUAL_RAIL, f'Q{i}'))
+            input_list[i * 2] = 1
+        default_input_state = BasicState(input_list)
+
+        for i, instruction in enumerate(qlmc.iterate_simple()):
+            instruction_name = instruction[0]  # name of the Gate
+            instruction_qbit = instruction[-1]  # tuple with list of qbit positions
+            # information carried by instruction
+            # tuple ('Name', [value of the parameter for gate], [list of qbit positions where gate is applied])
+
+            assert instruction_name in qlmc.gate_set, "cannot convert (%s)" % instruction_name
+            # only gates are converted -> checking if instruction is in gate_set of AQASM
+
+            if len(instruction_qbit) == 1:
+                if instruction_name == "H":
+                    ins = Circuit(2, name='H') // BS.H()
+                elif instruction_name == "PH":
+                    phi = instruction[1][0]  # value of the variable parameter in gate
+                    ins = Circuit(2, name='PS') // (1, PS(phi))  # apply phase shift on 2nd mode
+                else:
+                    gate_id = qlmc.ops[i].gate
+                    gate_matrix = qlmc.gateDic[gate_id].matrix  # gate matrix data
+                    gate_u = circ_to_np(gate_matrix)  # gate matrix to numpy
+                    ins = self._create_one_qubit_gate(gate_u)
+                p.add(instruction_qbit[0]*2, ins.copy())
+            else:
+                # only 2 qubit gates
+                c_idx = instruction_qbit[0] * 2
+                c_data = instruction_qbit[1] * 2
+                c_first = min(c_idx, c_data)  # used in SWAP
+
+                if instruction_name == "CNOT":
+                    cnot_idx += 1
+                    if use_postselection and cnot_idx == n_cnot:
+                        cnot_processor = self._postprocessed_cnot_builder.build()
+                        mode_map = {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+                    else:
+                        cnot_processor = self._heralded_cnot_builder.build()
+                        mode_map = {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+                    p.add(mode_map, cnot_processor)
+                elif instruction_name == "CSIGN":
+                    # Controlled Z in myqlm is named CSIGN
+                    cz_processor = self._heralded_cz_builder.build()
+                    mode_map = {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
```

Comment:
`mode_map` is build 3 times with the exact same pattern (2 if you take in account my comments above), consider using a tiny method to build it:
```
def get_mode_map(c_idx: int, c_data: int) -> dict:
    return {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
```

Reply:
Implemented with a function outside the class in the script